### ISSUE TITLE
Guidance needed for loss function and region/affinity maps

### ISSUE THREAD

**Issue by ghost** [clovaai/CRAFT-pytorch#157]
Hi, I'm currently developing a tensorflow equivalent of your code, I need guidance to calculate loss function properly and other necessary tips you may have for my version to work as expected.

Here's keras version of CRAFT, let me know if there's anything that needs to be fixed / modified:

    from tensorflow.keras import Model
    from tensorflow.keras.layers import (BatchNormalization, Concatenate, Conv2D,
                                         Input, MaxPooling2D, UpSampling2D,
                                         ZeroPadding2D)
    
    
    def create_model(
        input_shape=(224, 224, 3),
        activation='relu',
        padding='same',
        epsilon=1e-5,
        momentum=0.1,
    ):
        x0 = Input(input_shape)
        x = Conv2D(64, (3, 3), padding=padding, activation=activation)(x0)
        x = BatchNormalization(epsilon=epsilon, momentum=momentum)(x)
        x = Conv2D(64, (3, 3), padding=padding, activation=activation)(x)
        x = BatchNormalization(epsilon=epsilon, momentum=momentum)(x)
        x = MaxPooling2D(strides=(2, 2))(x)
        x = Conv2D(128, (3, 3), padding=padding, activation=activation)(x)
        x = BatchNormalization(epsilon=epsilon, momentum=momentum)(x)
        x = Conv2D(128, (3, 3), padding=padding, activation=activation)(x)
        s1 = BatchNormalization(epsilon=epsilon, momentum=momentum)(x)
        x = MaxPooling2D(strides=(2, 2))(s1)
        x = Conv2D(256, (3, 3), padding=padding, activation=activation)(x)
        x = BatchNormalization(epsilon=epsilon, momentum=momentum)(x)
        x = Conv2D(256, (3, 3), padding=padding, activation=activation)(x)
        s2 = BatchNormalization(epsilon=epsilon, momentum=momentum)(x)
        x = Conv2D(256, (3, 3), padding=padding, activation=activation)(s2)
        x = BatchNormalization(epsilon=epsilon, momentum=momentum)(x)
        x = MaxPooling2D(strides=(2, 2))(x)
        x = Conv2D(512, (3, 3), padding=padding, activation=activation)(x)
        x = BatchNormalization(epsilon=epsilon, momentum=momentum)(x)
        x = Conv2D(512, (3, 3), padding=padding, activation=activation)(x)
        s3 = BatchNormalization(epsilon=epsilon, momentum=momentum)(x)
        x = Conv2D(512, (3, 3), padding=padding, activation=activation)(s3)
        x = BatchNormalization(epsilon=epsilon, momentum=momentum)(x)
        x = MaxPooling2D(strides=(2, 2))(x)
        x = Conv2D(512, (3, 3), padding=padding, activation=activation)(x)
        x = BatchNormalization(epsilon=epsilon, momentum=momentum)(x)
        x = Conv2D(512, (3, 3), padding=padding, activation=activation)(x)
        s4 = BatchNormalization(epsilon=epsilon, momentum=momentum)(x)
        x = MaxPooling2D((3, 3), strides=(1, 1), padding='same')(s4)
        x = ZeroPadding2D((6, 6))(x)
        x = Conv2D(1024, (3, 3), dilation_rate=(6, 6))(x)
        s5 = Conv2D(1024, (1, 1))(x)
        x = Concatenate()([s5, s4])
        x = Conv2D(512, (1, 1), activation=activation)(x)
        x = BatchNormalization(epsilon=epsilon, momentum=momentum)(x)
        x = Conv2D(256, (3, 3), padding=padding, activation=activation)(x)
        x = UpSampling2D(interpolation='bilinear')(x)
        x = Concatenate()([x, s3])
        x = Conv2D(256, (1, 1), activation=activation)(x)
        x = BatchNormalization(epsilon=epsilon, momentum=momentum)(x)
        x = Conv2D(128, (3, 3), padding=padding, activation=activation)(x)
        x = BatchNormalization(epsilon=epsilon, momentum=momentum)(x)
        x = UpSampling2D(interpolation='bilinear')(x)
        x = Concatenate()([x, s2])
        x = Conv2D(128, (1, 1), activation=activation)(x)
        x = BatchNormalization(epsilon=epsilon, momentum=momentum)(x)
        x = Conv2D(64, (3, 3), padding=padding, activation=activation)(x)
        x = BatchNormalization(epsilon=epsilon, momentum=momentum)(x)
        x = UpSampling2D(interpolation='bilinear')(x)
        x = Concatenate()([x, s1])
        x = Conv2D(64, (1, 1), activation=activation)(x)
        x = BatchNormalization(epsilon=epsilon, momentum=momentum)(x)
        x = Conv2D(32, (3, 3), padding=padding, activation=activation)(x)
        x = BatchNormalization(epsilon=epsilon, momentum=momentum)(x)
        x = Conv2D(32, (3, 3), padding=padding, activation=activation)(x)
        x = Conv2D(32, (3, 3), padding=padding, activation=activation)(x)
        x = Conv2D(16, (3, 3), padding=padding, activation=activation)(x)
        x = Conv2D(16, (1, 1), activation=activation)(x)
        x = Conv2D(2, (1, 1))(x)
        return Model(x0, x)
    
    
    if __name__ == '__main__':
        m = create_model()
        m.summary()
